rust-runtime/aws-smithy-client/src/hyper_ext.rs

```diff
@@ -304,19 +304,22 @@ where
 
 #[cfg(feature = "rustls")]
 impl<M, R> ClientBuilder<(), M, R> {
-    /// Connect to the service over HTTPS using Rustls.
-    pub fn rustls(self) -> ClientBuilder<Adapter<crate::conns::Https>, M, R> {
-        self.connector(Adapter::builder().build(crate::conns::https()))
+    /// Connect to the service over HTTPS using Rustls using dynamic dispatch.
+    pub fn rustls(self) -> ClientBuilder<DynConnector, M, R> {
+        self.connector(DynConnector::new(
+            Adapter::builder().build(crate::conns::https()),
+        ))
     }
 }
 
 #[cfg(feature = "native-tls")]
 impl<M, R> ClientBuilder<(), M, R> {
-    /// Connect to the service over HTTPS using the native TLS library on your platform.
-    pub fn native_tls(
-        self,
-    ) -> ClientBuilder<Adapter<hyper_tls::HttpsConnector<hyper::client::HttpConnector>>, M, R> {
-        self.connector(Adapter::builder().build(crate::conns::native_tls()))
+    /// Connect to the service over HTTPS using the native TLS library on your
+    /// platform using dynamic dispatch.
+    pub fn native_tls(self) -> ClientBuilder<DynConnector, M, R> {
+        self.connector(DynConnector::new(
+            Adapter::builder().build(crate::conns::native_tls()),
+        ))
```

Comment:
I wonder if we should examples in these methods about how _not_ to use them when required (basically including examples of custom HTTPS, or at least linking to them)

Reply:
Filed an issue for improving these docs overall: https://github.com/awslabs/aws-sdk-rust/issues/469